### ISSUE TITLE
View shifts when usingt CTRL+F, then entering text while using a custom toolbar without find/search controls

### ISSUE THREAD

**Issue by Hollow-Ego** [stephanrauh/ngx-extended-pdf-viewer#2548]
Hi there, hello,

we noticed a bug when using a custom toolbar without any find/search controls. 

## Description
If the viewer doesn't have any find/search controls (find button, search input field, etc.) and the user tries to use the search, the viewer shifts to the left. On smaller screens (e.g. Full HD) that makes it look like the viewer disappeared and cannot be brought back without reloading the page. On larger screens (e.g. 4k) the shift is noticeable, but the viewer is still partially visible.

This happens in versions 20.5.0 and  21.3.8.

## General Steps to Reproduce
- Setup a  custom toolbar without including any options for search
- Open the app with the viewer and load file (if not loaded already)
- Hit Ctrl+F; nothing seems to happen
- Enter any letter(s); the view shifts

## Setup
This bug is also reproducible on the [custom toolbar demo](https://pdfviewer.net/extended-pdf-viewer/custom-toolbar) when selecting the "radiobutton zoom". 
For reference this is the template for it
```html
<ngx-extended-pdf-viewer
  #pdfViewer
  [customToolbar]="customCheckboxZoomToolbar"
  [src]="'/assets/pdfs/dachstein.pdf'"
  [textLayer]="true"
  [zoom]="zoom">
</ngx-extended-pdf-viewer>

<ng-template #customCheckboxZoomToolbar>
  <div id="toolbarViewer">
    <div [class.invisible]="false" id="toolbarViewerMiddle">
      <pdf-zoom-dropdown class="invisible"></pdf-zoom-dropdown>

      <input type="radio" id="zoom50" class="toolbarField radio" tabindex="94" (click)="zoom = '50%'" name="zoom" [checked]="zoom==='50%'" />
      <label for="zoom50" class="toolbarLabel">50%</label>

      <input type="radio" id="zoom100" class="toolbarField radio" tabindex="95" (click)="zoom = '100%'" name="zoom" [checked]="zoom==='100%'"/>
      <label for="zoom100" class="toolbarLabel">100%</label>

      <input type="radio" id="zoom200" class="toolbarField radio" tabindex="96" (click)="zoom = '200%'" name="zoom" [checked]="zoom==='200%'"/>
      <label for="zoom200" class="toolbarLabel">200%</label>

      <input type="radio" id="zoom-auto" class="toolbarField radio" tabindex="97" (click)="zoom = 'auto'" name="zoom" [checked]="zoom==='auto'"/>
      <label for="zoom-auto" class="toolbarLabel">auto</label>
    </div>
  </div>
</ng-template>
```

## Workarounds
As mentioned in #173 you can just entirely ignore CTRL+F 
```html
<ngx-extended-pdf-viewer
    [ignoreKeys]="['CTRL+F', 'CMD+F']"
```
If for some reason that's not what you want or can do, you can also provide a hidden dummy span with an id of `findInput`. This can be especially useful if you sometimes need the search and other times not.
```html
<span id="findInput" [hidden]="true"></span>
```

Unfortunately I wasn't able to find the responsible code to provide a solution. I could imagine having the option to explicitly disable the search functionality via a property.

**Comment by stephanrauh** [comment]
Interesting. I think the problem is that the find bar still exists - but it's shifted 3400 pixels to the right. I didn't expect that!
![image](https://github.com/user-attachments/assets/407aa0a4-b6ad-417f-82d6-1130cb16a6c9)

**Comment by stephanrauh** [comment]
BTW, thanks a lot for providing the work-arounds! That'll help other developers.

**Comment by Hollow-Ego** [comment]
> BTW, thanks a lot for providing the work-arounds! That'll help other developers.

You're welcome. That was the main goal. It's always frustrating to not find a good workaround and/or at least a hint of what is going on

**Comment by stephanrauh** [comment]
You've found a bug that's already been lurking four years in the code! So I think I'm allowed to change this, although strictly speaking, it's a breaking change...

Just in case you're curious, here's the bug:
```ts
this.findbarLeft = 34 + (32 * factor).toString() + 'px';
```
The opening parenthesis is at the wrong position, hence 34+32 is not 66, but 3432.

**Comment by stephanrauh** [comment]
I've added to changes (plus the bugfix):
- If `AppOptions.get("textLayerMode")` equals `TextLayerMode.DISABLE`, CTRL+F is suppressed. If I'm not mistaken, this covers documents without text layer. For these documents, finding doesn't work anyways.
- If the toolbar doesn't have a button with id `viewFind`, CTRL+F is also suppressed. Maybe there's a custom find button, but in this case, there's probably also a custom find bar.

**Comment by Hollow-Ego** [comment]
>  hence 34+32 is not 66, but 3432
Ah good ol' JavaScript. Glad this helped though

Your changes sound reasonable to me

**Comment by stephanrauh** [comment]
Your bug fix has landed with version 21.4.0.

Enjoy!
Stephan
